spec/models/import/actual_history_spec.rb

```diff
@@ -235,5 +235,36 @@
         expect(subject.errors.first.message).to include("Unknown RODA identifier")
       end
     end
+
+    context "when the row data has and incorrect but valid activity" do
+      let(:other_activity) { create(:project_activity) }
+      let(:row) {
+        CSV::Row.new(
+          ["RODA identifier", "Financial quarter", "Financial year", "Value"],
+          [other_activity.roda_identifier, "1", "2021", "100000"]
+        )
+      }
+
+      subject { described_class.new(report: @report, row_number: 2, row: row) }
+
+      it "returns false" do
+        expect(subject.call).to be false
+      end
+
+      it "returns errors" do
+        subject.call
+        expect(subject.errors.count).to eq 1
+      end
+
+      it "does not create the actual" do
+        expect(subject.actual).to be_nil
+      end
+
+      it "has a helpful error message" do
+        subject.call
+        expect(subject.errors.first.message)
+          .to include("does not match the fund and organisation of this report")
+      end
+    end
```

Comment:
did you consider merging this commit into the earlier "New Actual history importer"? Did you feel that it was important to show this change in direction? If this is strictly a "fix" to something which was wrong in the earlier commit, then one might take the view that it's clearer to only show a corrected version of history and do away with the mis-step in the earlier commit...

Reply:
It's a fair comment. I think by the time I reached here and really thougt about what would constitue a 'valid' report, I couldn't face the fixup - pure laziness! Let me see what I can do before merging this…